src/Illuminate/Foundation/Exceptions/Handler.php

```diff
@@ -220,12 +218,18 @@ protected function prepareJsonResponse($request, Exception $e)
 
         $headers = $this->isHttpException($e) ? $e->getHeaders() : [];
 
-        return response(json_encode([
-            'message' => $e->getMessage(),
-            'file' => $e->getFile(),
-            'line' => $e->getLine(),
-            'trace' => $e->getTrace(),
-        ], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES), $status, array_merge($headers, [
+        if (config('app.debug')) {
+            $response = [
+                'message' => $e->getMessage(),
+                'file' => $e->getFile(),
+                'line' => $e->getLine(),
+                'trace' => $e->getTrace(),
```

Comment:
> Sorry, I don't think we should response a different json in debug mode and the production mode.

Taylor already added JSON stuff for debug in the master branch. This PR is not adding it.

We are just putting JSON for production when it was requested via AJAX.